lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/OperatorTopology.cs

```diff
@@ -356,6 +393,72 @@ public bool HasChildren()
         }
 
         /// <summary>
+        /// Get a set of node containing an incoming message and belonging to candidate set of nodes.
+        /// </summary>
+        ///<param name="nodeSet">Candidate set of nodes</param>
+        /// <returns>A NodeStruct with incoming data.</returns>
+        /// private IEnumerable<NodeStruct> GetNodeWithData(IEnumerable<string> nodeSet)
+        private IEnumerable<NodeStruct> GetNodeWithData(IEnumerable<string> nodeSetIdentifier)
+        {
+            CancellationTokenSource timeoutSource = new CancellationTokenSource(_timeout);
+            List<NodeStruct> nodesSubsetWithData = new List<NodeStruct>();
+     
+            try
+            {
+                lock (_thisLock)
+                {
+                    foreach (var identifier in nodeSetIdentifier)
+                    {
+                        if (_idToNodeMap[identifier].HasMessage())
```

Comment:
If the identifier is not in the map, you would get index exception from _idToNodeMap[identifier].